manim/mobject/svg/text_mobject.py

```diff
@@ -1056,11 +1037,441 @@ def text2svg(self):
+            ].set_color(self._parse_color(col["color"]))
+        for grad in gradientmap:
+            self.chars[
+                grad["start"]
+                - grad["start_offset"] : grad["end"]
+                - grad["start_offset"]
+                - grad["end_offset"]
+            ].set_color_by_gradient(
+                *(self._parse_color(grad["from"]), self._parse_color(grad["to"]))
+            )
+        # anti-aliasing
+        if self.height is None and self.width is None:
+            self.scale(TEXT_MOB_SCALE_FACTOR)
+
+    def text2hash(self):
+        """Internally used function.
+        Generates ``sha256`` hash for file name.
+        """
+        settings = (
+            "MARKUPPANGO" + self.font + self.slant + self.weight
+        )  # to differentiate from classical Pango Text
+        settings += str(self.line_spacing) + str(self.size)
+        settings += str(self.disable_ligatures)
+        id_str = self.text + settings
+        hasher = hashlib.sha256()
+        hasher.update(id_str.encode())
+        return hasher.hexdigest()[:16]
+
+    def text2svg(self):
+        """Internally used function.
+        Convert the text to SVG using Pango
+        """
+        size = self.size * 10
+        line_spacing = self.line_spacing * 10
+        dir_name = config.get_dir("text_dir")
+        disable_liga = self.disable_ligatures
+        if not os.path.exists(dir_name):
+            os.makedirs(dir_name)
+        hash_name = self.text2hash()
+        file_name = os.path.join(dir_name, hash_name) + ".svg"
+        if os.path.exists(file_name):
+            return file_name
+        surface = cairocffi.SVGSurface(file_name, 600, 400)
+        context = cairocffi.Context(surface)
+        context.move_to(START_X, START_Y)
+        layout = pangocairocffi.create_layout(context)
+        layout.set_width(pangocffi.units_from_double(600))
+
+        fontdesc = pangocffi.FontDescription()
+        fontdesc.set_size(pangocffi.units_from_double(size))
+        if self.font:
+            fontdesc.set_family(self.font)
+        fontdesc.set_style(PangoUtils.str2style(self.slant))
+        fontdesc.set_weight(PangoUtils.str2weight(self.weight))
+        layout.set_font_description(fontdesc)
+
+        context.move_to(START_X, START_Y)
+        pangocairocffi.update_layout(context, layout)
+        if disable_liga:
+            layout.set_markup(
+                f"<span font_features='liga=0,dlig=0,clig=0,hlig=0'>{self.text}</span>"
+            )
+        else:
+            layout.set_markup(self.text)
+        logger.debug(f"Setting Text {self.text}")
+        pangocairocffi.show_layout(context, layout)
+        surface.finish()
+        return file_name
+
+    def _count_real_chars(self, s):
+        """Internally used function.
+        Counts characters that will be displayed.
+        This is needed for partial coloring or gradients, because space
+        counts to the text's `len`, but has no corresponding letter"""
```

Comment:
```suggestion
        """Counts characters that will be displayed.

        This is needed for partial coloring or gradients, because space
        counts to the text's `len`, but has no corresponding character."""
```